homeassistant/components/watergate/config_flow.py

```diff
@@ -0,0 +1,73 @@
+"""Config flow for Watergate."""
+
+from typing import Any
+
+import voluptuous as vol
+from watergate_local_api.watergate_api import WatergateLocalApiClient
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_IP_ADDRESS, CONF_NAME
+
+from .const import DOMAIN
+
+WATERGATE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): str,
+        vol.Required(CONF_IP_ADDRESS): str,
+    }
+)
+
+
+class WatergateConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Watergate config flow."""
+
+    async def async_step_user(
+        self, user_input: dict[str, str] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initiated by the user."""
+        if user_input is not None and await self.validate_connection(
```

Comment:
if ``validate_connection` evaluates to `false`, we should show an error (by adding to the `errors` param ofs how_form)